marshal.go

```diff
@@ -1379,3 +1378,22 @@ func (x *GoSNMP) receive() ([]byte, error) {
 	copy(resp, x.rxBuf[:n])
 	return resp, nil
 }
+
+func shrinkAndWriteUint(buf io.Writer, id uint) error {
+	var pktid any
+	switch {
+	case id < math.MaxUint8:
```

Comment:
Should these be `<=`? For example, `255` should fit in 1 byte, but this would make it stored in 2 bytes.